guppylang/std/qsystem/random.py

```diff
@@ -0,0 +1,105 @@
+from typing import no_type_check
+
+from hugr import Wire
+from hugr import tys as ht
+from hugr.std.int import int_t
+
+from guppylang.decorator import guppy
+from guppylang.definition.custom import CustomInoutCallCompiler
+from guppylang.definition.value import CallReturnWires
+from guppylang.module import GuppyModule
+from guppylang.std._internal.compiler.arithmetic import inarrow_s, iwiden_s
+from guppylang.std._internal.compiler.prelude import build_unwrap_right
+from guppylang.std._internal.compiler.quantum import (
+    QSYSTEM_RANDOM_EXTENSION,
+    RNGCONTEXT_T,
+)
+from guppylang.std._internal.util import external_op
+from guppylang.std.builtins import nat, owned
+from guppylang.std.option import Option
+
+qsystem_random = GuppyModule("qsystem.random")
+
+
+class RandomIntCompiler(CustomInoutCallCompiler):
```

Comment:
Could you move those `CustomInoutCallCompiler`s into the `_internal` module.

Maybe into `guppylang.std._internal.compiler.quantum` or a new `guppylang.std._internal.compiler.qsystem`?